miden-lib/asm/miden/contracts/wallets/basic.masm

```diff
@@ -73,3 +73,165 @@ export.send_asset.1
     # prepare the stack for return - stack has 6 elements too many
     movupw.3 dropw swap drop swap drop
 end
+
+#! Creates a note which sends the specified amount of assets out of the current account to the 
+#! specified recipient. The key difference with the `send_asset` procedure is that `send_note` 
+#! allows to create notes without assets whatsoever.
+#!
+#! Inputs stack: [tag, aux, note_type, RECIPIENT, ASSETS_HASH, ...]
+#! Advice stack: [num_assets_felt, [ASSETS]]
+#! Outputs: [note_idx, 0, 0, EMPTY_WORD, ...]
+#!
+#! - tag is the tag to be included in the note.
+#! - aux is the auxiliary data to be included in the note.
+#! - note_type is the note's storage type
+#! - RECIPIENT is the recipient of the note, i.e.,
+#!   hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)
+#! - ASSETS_HASH is the sequential hash of the assets which are going to be added to the note.
+#! - num_assets_felt is the number of Felts which constitute the assets to be added to the note.
+#! - [ASSETS] is an array of the assets which are going to be added to the note.
+#!
+#! Panics:
+#! - The fungible asset is not found in the vault.
+#! - The amount of the fungible asset in the vault is less than the amount to be removed.
+#! - The non-fungible asset is not found in the vault.
+export.send_note
+    ### Create the note #######################################################
+    
+    # This procedure is written to be executed with `exec` or `call`. When this
+    # procedure is `call`ed the stack has to be carefully manipulated to avoid
+    # inserting unwanted elements between the user data. The convention is to
+    # ensure the input&output data have the same length. This code pads the
+    # stack so the output stack will be the same length as the input.
+    #
+    # The padding must be added before calling `create_note`, not after. This
+    # is because the VM stack has a minimum size of 16 elements, trying to push
+    # elements after the call to `create_note` would increase the stack in
+    # addition to the minimum 16 elements.
+    push.0 movdn.11 padw movdnw.3
+    # => [tag, aux, note_type, RECIPIENT, ASSETS_HASH, 0, EMPTY_WORD, ...]
```

Comment:
Let's change the assumption here: let's assume that this procedure will always be `call`-ed. Also, let's explicitly say that the procedure takes 16 stack elements and returns 16 stack elements (the extra elements padded with 0s) - kind of alone the lines what is described in https://github.com/0xPolygonMiden/miden-base/issues/685.